client/src/app/mainwindow.cpp

```diff
@@ -680,6 +681,18 @@ void MainWindow::initialResourceSelection()
         mLoadingOverlay->setMessage(i18n("Configure a SugarCRM resource in order to use FatCRM."));
         showResourceDialog();
     } else {
+        QString defaultResourceId = ClientSettings::self()->defaultResourceId();
+        if (!defaultResourceId.isEmpty()) {
+            for (int index = 0; index < mResourceSelector->count(); ++index) {
+                const AgentInstance agent = mResourceSelector->itemData(index, AgentInstanceModel::InstanceRole).value<AgentInstance>();
+                if (agent.isValid() && (agent.identifier() == defaultResourceId)) {
+                    slotResourceSelectionChanged(index);
```

Comment:
recursive call? urgh. Surely this can be done by extracting the code that we want to run here.